Engines/Wine/Verbs/DXVK/script.js

```diff
@@ -9,14 +9,19 @@ include(["utils", "functions", "filesystem", "files"]);
 */
 Wine.prototype.DXVK = function () {
     print("NOTE: you need a driver that supports Vulkan enough to run DXVK");
-    print("NOTE: wine version should be greater or equal to 3.5");
-
-    var dxvkVersion = "0.53";
+    print("NOTE: wine version should be greater or equal to 3.10");
+    
+    var releaseFile = new Resource()
+        .wizzard(this.wizzard())
+        .url("https://github.com/doitsujin/dxvk/blob/master/RELEASE")
+        .name("RELEASE.txt")
+        .get();
+    
+    var dxvkVersion = function cat(this.prefixDirectory() + "RELEASE.txt");
```

Comment:
Did you actually test this? This line looks very strange to me. I'm not even sure this is valid Javascript.

Reply:
I thought it may work based on what I read about `cat` function but I get this error:
```
org.phoenicis.scripts.interpreter.ScriptException: javax.script.ScriptException: <eval>:21:35 Expected ident but found this
    var dxvkVersion = function cat(this.prefixDirectory() + "RELEASE.txt");
                                   ^ in <eval> at line number 21 at column number 35
	at org.phoenicis.scripts.nashorn.builtins.EngineInjector.throwException(EngineInjector.java:22)
	at org.phoenicis.scripts.nashorn.NashornEngine.handleError(NashornEngine.java:74)
	at org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:53)
	at org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:44)
	at org.phoenicis.scripts.nashorn.builtins.IncludeInjector.lambda$injectInto$0(IncludeInjector.java:36)
	at jdk.nashorn.internal.scripts.Script$416$\^eval\_.:scopeCall(<eval>)
	at jdk.nashorn.internal.scripts.Script$416$\^eval\_.:program(<eval>:11)
	at jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:637)
	at jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:494)
	at jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:393)
	at jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:449)
	at jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:406)
	at jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:402)
	at jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:155)
	at javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
	at org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:50)
	at org.phoenicis.scripts.nashorn.NashornScriptInterpreter.runScript(NashornScriptInterpreter.java:35)
	at org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:35)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: javax.script.ScriptException: <eval>:21:35 Expected ident but found this
    var dxvkVersion = function cat(this.prefixDirectory() + "RELEASE.txt");
                                   ^ in <eval> at line number 21 at column number 35
	at jdk.nashorn.api.scripting.NashornScriptEngine.throwAsScriptException(NashornScriptEngine.java:470)
	at jdk.nashorn.api.scripting.NashornScriptEngine.compileImpl(NashornScriptEngine.java:537)
	at jdk.nashorn.api.scripting.NashornScriptEngine.compileImpl(NashornScriptEngine.java:524)
	at jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:402)
	at jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:155)
	at javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
	at org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:50)
	... 18 more
Caused by: jdk.nashorn.internal.runtime.ParserException: <eval>:21:35 Expected ident but found this
    var dxvkVersion = function cat(this.prefixDirectory() + "RELEASE.txt");
                                   ^
	at jdk.nashorn.internal.parser.AbstractParser.error(AbstractParser.java:294)
	at jdk.nashorn.internal.parser.AbstractParser.error(AbstractParser.java:279)
	at jdk.nashorn.internal.parser.AbstractParser.expectValue(AbstractParser.java:363)
	at jdk.nashorn.internal.parser.AbstractParser.getIdent(AbstractParser.java:431)
	at jdk.nashorn.internal.parser.Parser.formalParameterList(Parser.java:2835)
	at jdk.nashorn.internal.parser.Parser.formalParameterList(Parser.java:2804)
	at jdk.nashorn.internal.parser.Parser.functionExpression(Parser.java:2655)
	at jdk.nashorn.internal.parser.Parser.memberExpression(Parser.java:2506)
	at jdk.nashorn.internal.parser.Parser.leftHandSideExpression(Parser.java:2372)
	at jdk.nashorn.internal.parser.Parser.unaryExpression(Parser.java:3147)
	at jdk.nashorn.internal.parser.Parser.assignmentExpression(Parser.java:3353)
	at jdk.nashorn.internal.parser.Parser.variableStatement(Parser.java:1088)
	at jdk.nashorn.internal.parser.Parser.statement(Parser.java:884)
	at jdk.nashorn.internal.parser.Parser.sourceElements(Parser.java:773)
	at jdk.nashorn.internal.parser.Parser.functionBody(Parser.java:2901)
	at jdk.nashorn.internal.parser.Parser.functionExpression(Parser.java:2663)
	at jdk.nashorn.internal.parser.Parser.memberExpression(Parser.java:2506)
	at jdk.nashorn.internal.parser.Parser.leftHandSideExpression(Parser.java:2372)
	at jdk.nashorn.internal.parser.Parser.unaryExpression(Parser.java:3147)
	at jdk.nashorn.internal.parser.Parser.expression(Parser.java:3325)
	at jdk.nashorn.internal.parser.Parser.expression(Parser.java:3282)
	at jdk.nashorn.internal.parser.Parser.expressionStatement(Parser.java:1150)
	at jdk.nashorn.internal.parser.Parser.statement(Parser.java:967)
	at jdk.nashorn.internal.parser.Parser.sourceElements(Parser.java:773)
	at jdk.nashorn.internal.parser.Parser.program(Parser.java:709)
	at jdk.nashorn.internal.parser.Parser.parse(Parser.java:283)
	at jdk.nashorn.internal.parser.Parser.parse(Parser.java:249)
	at jdk.nashorn.internal.runtime.Context.compile(Context.java:1284)
	at jdk.nashorn.internal.runtime.Context.compileScript(Context.java:1251)
	at jdk.nashorn.internal.runtime.Context.compileScript(Context.java:627)
	at jdk.nashorn.api.scripting.NashornScriptEngine.compileImpl(NashornScriptEngine.java:535)
	... 23 more
```

Reply:
Why is there "function" before "cat" ?

Reply:
I'm used to C++ style.

Reply:
I don't see how this should be any different in C++. Anyways: I would really like to ask you to test your changes before providing a pull request. The reviewers cannot always spot this kind of problem.

Reply:
This is working (and do version choosing)
```js
/**
* Setup DXVK-> https://github.com/doitsujin/dxvk/
* @param dxvkVersion DXVK version to download
* @returns {Wine} Wine object
*/
Wine.prototype.DXVK = function (dxvkVersion) {
    print("NOTE: you need a driver that supports Vulkan enough to run DXVK");
    print("NOTE: wine version should be greater or equal to 3.10");

    if(typeof dxvkVersion === 'undefined')
    {
        var releaseFile = new Resource()
            .wizard(this.wizard())
            .url("https://raw.githubusercontent.com/doitsujin/dxvk/master/RELEASE")
            .name("RELEASE.txt")
            .get();
    
        dxvkVersion = cat(releaseFile);
    }
```
The problem is that the release file contains something like a '\n' character which break the download afterwards.
I tried to do this ->[https://stackoverflow.com/questions/10805125/how-to-remove-all-line-breaks-from-a-string]( https://stackoverflow.com/questions/10805125/how-to-remove-all-line-breaks-from-a-string) but it did not work. @plata Do you have an idea about this ?

Reply:
What do you want to do with the `\n`?
Do you only want to remove them or replace them with something else?
In case you want to remove them, you can use https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#replaceAll(java.lang.String,%20java.lang.String):
```
dxvkVersion = cat(releaseFile).replaceAll("\\n", "");
```

Reply:
Ok, i forgot the double \... Now it is working, thanks.

@Zemogiter you should be able to finalize this now ;-)